src/modules/ekf2/EKF/ekf_helper.cpp

```diff
@@ -695,40 +695,62 @@ bool Ekf::getEkfGlobalOrigin(uint64_t &origin_time, double &latitude, double &lo
 	origin_time = _last_gps_origin_time_us;
 	latitude = _pos_ref.getProjectionReferenceLat();
 	longitude = _pos_ref.getProjectionReferenceLon();
-	origin_alt  = _gps_alt_ref;
+	origin_alt  = getEkfGlobalOriginAltitude();
 	return _NED_origin_initialised;
 }
 
-void Ekf::setEkfGlobalOrigin(const double latitude, const double longitude, const float altitude)
+bool Ekf::setEkfGlobalOrigin(const double latitude, const double longitude, const float altitude)
 {
-	bool current_pos_available = false;
-	double current_lat = static_cast<double>(NAN);
-	double current_lon = static_cast<double>(NAN);
-	float current_alt  = 0.f;
+	// sanity check valid latitude/longitude and altitude anywhere between the Mariana Trench and edge of Space
+	if (PX4_ISFINITE(latitude) && (abs(latitude) <= 90)
+	&& PX4_ISFINITE(longitude) && (abs(latitude) <= 180)
+	&& PX4_ISFINITE(altitude) && altitude > -12000.f && altitude < 100'000.f
```

Comment:
In case this is new to anyone else, I was unaware of how the compiler handles apostrophes in numbers until now. https://stackoverflow.com/questions/54447208/why-can-i-separate-digits-with-an-apostrophe-in-c-but-not-c